relay-server/src/services/store.rs

```diff
@@ -1418,8 +1419,8 @@ struct SpanKafkaMessage<'a> {
     #[serde(default)]
     is_remote: bool,
 
-    #[serde(default, skip_serializing_if = "none_or_empty_object")]
-    data: Option<&'a RawValue>,
+    #[serde(default, skip_serializing_if = "none_or_empty_map")]
+    data: Option<BTreeMap<String, Option<serde_json::Value>>>,
```

Comment:
Can we deserialize the value into something that borrows?

Without having it checked, having the `data` and `sentry_tags` values deserialized as `&'a RawValue` with a `#[serde(borrow)]` should get around all allocations.